org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/markers/MarkerUtils.java

```diff
@@ -177,11 +180,20 @@ public long getMarkerId() {
       return markerId;
     }
 
+    @CheckForNull
+    public ExtraPosition getParent() {
+      return previous;
+    }
+
     @Override
     public boolean equals(Object other) {
       if (other instanceof ExtraPosition) {
         ExtraPosition rp = (ExtraPosition) other;
-        return (rp.offset == offset) && (rp.length == length) && (rp.markerId == markerId) && Objects.equals(rp.message, message);
+        return (rp.offset == offset)
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1067)